javascript/ql/lib/semmle/javascript/frameworks/ShellJS.qll

```diff
@@ -1,15 +1,29 @@
 /**
  * Models the `shelljs` library in terms of `FileSystemAccess` and `SystemCommandExecution`.
+ *
+ * https://www.npmjs.com/package/shelljs
  */
 
 import javascript
 
 module ShellJS {
+  API::Node shellJSMember() {
+    result = API::moduleImport("shelljs")
+    or
+    result =
+      shellJSMember()
+          .getMember([
+              "exec", "cd", "cp", "touch", "chmod", "pushd", "find", "ls", "ln", "mkdir", "mv",
+              "rm", "cat", "head", "sort", "tail", "uniq", "grep", "sed", "to", "toEnd", "echo"
+            ])
```

Comment:
I took a look at the type-definition of ShellJS, and from that documentation it appears that only some of the functions behave the way you have modelled here.  
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/shelljs/index.d.ts#L830 

However, looking at the implementation (and specifically [this PR](https://github.com/shelljs/shelljs/pull/934/files)), it seems that the type is misleading, and that your model is correct.